src/Server/modules/USPlacesManager.js

```diff
@@ -0,0 +1,331 @@
+const dataClassesModule = require('../modules/DataClasses.js');
+const readCSVModule = require('../modules/ReadCSV.js');
+const writeCSVModule = require('../modules/WriteCSV.js');
+
+var allUSPlaces = [];
+
+function InitializeAllUSPlace(){
+    allUSPlaces = readCSVModule.LoadUSData();
+}// read allUSdata and put into array
+
+function AddUSData(county, state, date, tod, number) {
+    
+    if(allUSPlaces == undefined){
+        InitializeAllUSPlace();
+    }
+
+	var errormsg = 'no error';
+	if (tod == 1) {
+		for (var i = 0; i < allUSPlaces.length; ++i) {
+			if (county == allUSPlaces[i].county && state == allUSPlaces[i].state) {
+				for (var key in allUSPlaces[i].numConfirmed) {
+					var temp_date = new Date(key);
+					var d =
+						temp_date.getMonth() +
+						1 +
+						'/' +
+						temp_date.getDate() +
+						'/' +
+						temp_date.getFullYear();
+					if (d == date) {
+						console.log('ERROOOOOOOR');
+						//return result;
+						errormsg = 'date exists';
+						return errormsg;
+					}
+				}
+				var temp_date = helper.stringToDate(date);
+				allUSPlaces[i].numDeaths[temp_date] = number;
+				if(allUSPlaces[i].currentNumDeaths!=number){
+					allUSPlaces[i].currentNumDeaths=number;
+				}
+				break;
+			} else if (i == allUSPlaces.length - 1) {
+				// if its the last element & prog didn't go into the 1st if -> place doesn't exist
+				errormsg = 'wrong place';
+			}
+		}
+	} else if (tod == 2) {
+		for (var i = 0; i < allUSPlaces.length; ++i) {
+			if (county == allUSPlaces[i].county && state == allUSPlaces[i].state) {
+				for (var key in allUSPlaces[i].numDeaths) {
+					var temp_date = new Date(key);
+					var d =
+						temp_date.getMonth() +
+						1 +
+						'/' +
+						temp_date.getDate() +
+						'/' +
+						temp_date.getFullYear();
+					if (d == date) {
+						console.log('ERROOOOOOOR');
+						errormsg = 'date exists';
+						return errormsg;
+					}
+				}
+				var temp_date = helper.stringToDate(date);
+				allUSPlaces[i].numConfirmed[temp_date] = number;
+				if(allUSPlaces[i].currentNumConfirmed!=number){
```

Comment:
We don't want to assign currentNumConfirmed = number here. If currentNumConfirmed = 26, because there are 26 cases on 5/10/20, and the user tries to add the data {5/1/20, 17}, we want to keep currentNumConfirmed = 26, because 5/10/20 is more recent than 5/1/20.

Instead, we can do the same thing as we did in the DataClasses.js addNumDeaths(). We can go through all elements in the dictionary to find the data for the most recent date.